examples/UNet/massachussets_road_segm.py

```diff
@@ -0,0 +1,340 @@
+__author__ = 'Fabian Isensee'
+import numpy as np
+import lasagne
+import os
+import sys
+import fnmatch
+import matplotlib.pyplot as plt
+sys.path.append("../../modelzoo/")
+from Unet import *
+import theano.tensor as T
+import theano
+import cPickle
+from time import sleep
+
+def prep_folders():
+    if not os.path.isdir("data"):
+        os.mkdir("data")
+
+    if not os.path.isdir("data/validation"):
+        os.mkdir("data/validation")
+    if not os.path.isdir("data/training"):
+        os.mkdir("data/training")
+    if not os.path.isdir("data/test"):
+        os.mkdir("data/test")
+
+    if not os.path.isdir("data/validation/sat_img"):
+        os.mkdir("data/validation/sat_img")
+    if not os.path.isdir("data/validation/map"):
+        os.mkdir("data/validation/map")
+    if not os.path.isdir("data/training/sat_img"):
+        os.mkdir("data/training/sat_img")
+    if not os.path.isdir("data/training/map"):
+        os.mkdir("data/training/map")
+    if not os.path.isdir("data/test/sat_img"):
+        os.mkdir("data/test/sat_img")
+    if not os.path.isdir("data/test/map"):
+        os.mkdir("data/test/map")
+
+def prep_urls():
+    valid_data_url = valid_target_url = np.loadtxt("mass_roads_validation.txt", dtype=str)
+    valid_data_str = "https://www.cs.toronto.edu/~vmnih/data/mass_roads/valid/sat/"
+    valid_target_str = "https://www.cs.toronto.edu/~vmnih/data/mass_roads/valid/map/"
+
+    train_data_url = train_target_url  = np.loadtxt("mass_roads_train.txt", dtype=str)
+    train_data_str = "https://www.cs.toronto.edu/~vmnih/data/mass_roads/train/sat/"
+    train_target_str = "https://www.cs.toronto.edu/~vmnih/data/mass_roads/train/map/"
+
+    test_data_url = test_target_url  = np.loadtxt("mass_roads_test.txt", dtype=str)
+    test_data_str = "https://www.cs.toronto.edu/~vmnih/data/mass_roads/test/sat/"
+    test_target_str = "https://www.cs.toronto.edu/~vmnih/data/mass_roads/test/map/"
+
+    f = open("mass_roads_train_data_download.sh", 'w')
+    g = open("mass_roads_train_target_download.sh", 'w')
+    for img_name in train_data_url:
+        f.write("wget -O data/training/sat_img/%sf "%img_name + train_data_str + img_name + "f" + "\n")
+        g.write("wget -O data/training/map/%s "%img_name + train_target_str + img_name + "\n")
```

Comment:
Can't you download the files directly, using urlretrieve, similar to the MNIST example: https://github.com/Lasagne/Lasagne/blob/5a009f9/examples/mnist.py#L33-L41 ? The current solution won't work on Windows.
You can assemble a list of source URLs and target file names, create a `multiprocessing.dummy.Pool` (like `multiprocessing.Pool`, but using threads) and call its `map()` method to parallelize the downloads.